spec.md

```diff
@@ -179,17 +179,163 @@ physically present in the final circuit. Refer to the description of the
 instance statement for details on how to instantiate a module
 ([@sec:instances]).
 
+## Optional Groups
+
+Optional groups are named collections of statements inside a module.  Optional
+groups contain optional functionality for a module that can be included or
+excluded without affecting the functionality of a circuit.  Optional groups are
```

Comment:
I understand what you are going for, but maybe better wording?  I'm not sure quite what.  "Optional groups contain functionality which will not be present in all executions of a circuit."